components/data_comps/dlnd/src/lnd_comp_mct.F90

```diff
@@ -146,13 +152,25 @@ subroutine lnd_init_mct( EClock, cdata, x2l, l2x, NLFilename )
     !----------------------------------------------------------------------------
     ! Initialize dlnd
     !----------------------------------------------------------------------------
-
+#ifdef HAVE_MOAB
+    ierr = iMOAB_RegisterApplication(trim("DLND")//C_NULL_CHAR, mpicom, compid, mlnid)
+    if (ierr .ne. 0) then
+      write(logunit,*) subname,' error in registering data lnd comp'
+      call shr_sys_abort(subname//' ERROR in registering data lnd comp')
+    endif
+#endif
     call dlnd_comp_init(Eclock, x2l, l2x, &
          seq_flds_x2l_fields, seq_flds_l2x_fields, &
          SDLND, gsmap, ggrid, mpicom, compid, my_task, master_task, &
          inst_suffix, inst_name, logunit, read_restart, &
          scmMode, scmlat, scmlon)
-
+#ifdef HAVE_MOAB
+     if (my_task == master_task) then
+          call seq_infodata_PutData( infodata, lnd_domain=SDLND%domainFile) ! we use the same one for regular case
```

Comment:
Why is this needed only for MOAB driver?

Reply:
the regular coupler does not need the mesh; we do